ts/features/bonusVacanze/store/sagas/bonusSaga.ts

```diff
@@ -1,10 +1,126 @@
 import { readableReport } from "italia-ts-commons/lib/reporters";
+import { Millisecond } from "italia-ts-commons/lib/units";
 import { SagaIterator } from "redux-saga";
 import { call, put, takeLatest } from "redux-saga/effects";
 import { apiUrlPrefix } from "../../../../config";
 import { SagaCallReturnType } from "../../../../types/utils";
+import { startTimer } from "../../../../utils/timer";
 import { BackendBonusVacanze } from "../../api/backendBonusVacanze";
-import { availableBonusListLoad } from "../actions/bonusVacanze";
+import {
+  availableBonusesLoad,
+  checkBonusEligibility,
+  eligibilityRequestProgress
+} from "../actions/bonusVacanze";
+import { EligibilityRequestProgressEnum } from "../reducers/bonusVacanze";
+
+const checkEligibilityResultPolling = 1000 as Millisecond;
+// stop polling when elapsed time from the beginning exceeds this threshold
+const pollingTimeThreshold = (10 * 1000) as Millisecond;
+
+// handle start bonus eligibility check
+function* checkBonusEligibilitySaga(
+  getEligibilityCheck: ReturnType<
+    typeof BackendBonusVacanze
+  >["getEligibilityCheck"]
+): SagaIterator {
+  try {
+    const eligibilityCheckResult: SagaCallReturnType<
+      typeof getEligibilityCheck
+    > = yield call(getEligibilityCheck, {});
+    if (eligibilityCheckResult.isRight()) {
+      // we got the check result
+      if (eligibilityCheckResult.value.status === 200) {
+        yield put(
+          checkBonusEligibility.success(eligibilityCheckResult.value.value)
+        );
+        return true;
+      }
+      return false;
+    } else {
+      // we got some error, stop polling
+      throw Error(readableReport(eligibilityCheckResult.value));
+    }
+  } catch (e) {
+    yield put(checkBonusEligibility.failure(e));
+    return false;
+  }
+}
+
+// handle start bonus eligibility check
+// tslint:disable-next-line: cognitive-complexity
+function* startBonusEligibilitySaga(
+  postEligibilityCheck: ReturnType<
+    typeof BackendBonusVacanze
+  >["postEligibilityCheck"],
+  getEligibilityCheck: ReturnType<
+    typeof BackendBonusVacanze
+  >["getEligibilityCheck"]
+): SagaIterator {
+  try {
+    const startEligibilityResult: SagaCallReturnType<
+      typeof postEligibilityCheck
+    > = yield call(postEligibilityCheck, {});
+    if (startEligibilityResult.isRight()) {
+      // request accepted | pending request
+      if (
+        startEligibilityResult.value.status === 202 ||
+        startEligibilityResult.value.status === 409
+      ) {
+        // request is pending
+        yield put(
+          eligibilityRequestProgress(EligibilityRequestProgressEnum.PENDING)
+        );
```

Comment:
I think this should be dispatched at the start of the saga, before the first remote call, otherwise in the first phase we would be in an indefinite state.

Reply:
@fabriziofff yes it should be, my fault. Update within https://github.com/pagopa/io-app/pull/1849/commits/7a6ccd6c518932b69454ade02aeb299db02daf96